clients/src/main/java/org/apache/kafka/common/record/FileRecords.java

```diff
@@ -177,6 +181,16 @@ public void closeHandlers() throws IOException {
         channel.close();
     }
 
+    @Override
+    public void reopen(File f) {
+        if(OperatingSystem.IS_WINDOWS){
+            try {
+                channel = openChannel(f, true, true, this.start, false);
+            }
+            catch (IOException e){
```

Comment:
I agree with Bharat. We  This Exception needs to be either logged or rethrown. It cannot simply disappear here.